test/integration/kubernetes_deploy_test.rb

```diff
@@ -567,6 +569,25 @@ def test_success_detection_tolerates_out_of_band_deployment_scaling
     assert_equal true, success, "Failed to deploy deployment with dynamic replica count"
   end
 
+  def test_bad_container_on_daemon_sets_fails
+    success = deploy_fixtures("hello-cloud", subset: ["daemon_set.yml"]) do |fixtures|
+      daemon_set = fixtures['daemon_set.yml']['DaemonSet'].first
+      container = daemon_set['spec']['template']['spec']['containers'].first
+      container["image"] = "busybox"
+      container["command"] = ["ls", "/not-a-dir"]
+    end
+
+    refute success
+
+    assert_logs_match_all([
+      "DaemonSet/nginx: FAILED",
+      "The following containers are in a state that is unlikely to be recoverable:",
+      "Events: None found. Please check your usual logging service (e.g. Splunk).",
```

Comment:
Hm, really? I'd expect there to be some events here. Indeed, I see two when I run this test locally:
```
[Pod/nginx-cnkzq]	BackOff: Back-off restarting failed container (2 events)
[Pod/nginx-cnkzq]	FailedSync: Error syncing pod, skipping: failed to "StartContainer" for "nginx" with CrashLoopBackOff: "Back-off 10s restarting failed container=nginx pod=nginx-cnkzq_k8sdeploy-test-bad-container-on-daemon-sets-fa-be1043b5b97dfe2b(969dd56c-7152-11e7-be12-76b0ec54df39)" (2 events)
```